src/test/java/sirius/web/data/ExcelExportSpec.groovy

```diff
@@ -56,4 +58,38 @@ class ExcelExportSpec extends BaseSpecification {
         cleanup:
         Files.delete(testFile)
     }
+
+    def "only allow 1 million in a excel sheet rows"() {
+        given:
+        File testFile = File.createTempFile("excel-output", ".xlsx")
+        when:
+        ExcelExport export = ExcelExport.asStreamingXLSX()
+        StringBuilder sheetWithError = new StringBuilder()
+        export.setMaxRowsReachedHandler({ sheetName -> sheetWithError.append(sheetName) })
+        for (int i = 1; i <= ExcelExport.MAX_NUM_ROWS; i++) {
+            export.addRow("A-" + i)
+        }
+        export.addRow("A-" + (ExcelExport.MAX_NUM_ROWS + 1))
+        export.writeToStream(new FileOutputStream(testFile))
+        then:
+        sheetWithError.toString() == "Sheet0"
+        and:
+        Counter lineCounter = new Counter()
+        LineBasedProcessor.create(testFile.getName(), new FileInputStream(testFile))
+                          .run({
+                                   lineNum, row ->
+                                       lineCounter.inc()
+                                       assert lineNum <= ExcelExport.MAX_NUM_ROWS + 1
```

Comment:
assert? is that the spock way to do it? these might be disabled in the JVM, right?

Reply:
well I think so. it does not work without the assert.

here is an article describing what code is actually generated by spock, but the example is with an `if` an not with a closure
https://e.printstacktrace.blog/spock-assertion-inside-if-statement-doesnt-work-why/

Reply:
so the `assert` is just a hint for Spock to generate the default assertion code also generated by regular spock assertions in the `then:` block

Reply:
th,x good to know